### PR TITLE
Remove list of snaps a user is a collaborator on

### PR DESCRIPTION
**Pull request by Lukewh**
# To be rebased after https://github.com/canonical/snapcraft.io/pull/4286 is merged

## Done
- Remove fetching and storing *all* brand stores in the session

## How to QA
- Login to https://snapcraft-io-4287.demos.haus/
- Ensure brand store admin still works as before
- Go to your list of snaps
- Click "Register a snap name"
- You should see a dropdown with a list of available stores
- Attempt to register a name in one of the stores
- All screens you see in this flow should contain the dropdown with the list of stores you have publisher rights to.

## Issue / Card
Fixes https://github.com/canonical/snapcraft.io/issues/3838
Fixes https://github.com/canonical/snapcraft.io/issues/4277

### CODE REVIEW

File: static/js/base/navigation.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ if (navAccountContainer) {
         authenticatedMenu.classList.remove("u-hide");
         displayName.innerHTML = data.publisher["fullname"];
 
-        if (data.publisher.stores) {
+        if (data.publisher.has_stores) {
```
Comment (Lukewh) [inline static/js/base/navigation.js:24]: update to the new key

File: webapp/login/views.py [hunk 2]
```diff
@@ -94,8 +92,8 @@ def after_login(resp):
         if logic.get_stores(
             account["stores"], roles=["admin", "review", "view"]
         ):
-            flask.session["publisher"]["stores"] = admin_api.get_stores(
-                flask.session
+            flask.session["publisher"]["has_stores"] = (
```
Comment (Lukewh) [inline webapp/login/views.py:95]: get the stores and only store whether there are any

File: webapp/login/views.py [hunk 3]
```diff
@@ -94,8 +92,8 @@ def after_login(resp):
...
+                len(admin_api.get_stores(flask.session)) > 0
```
Comment (Lukewh) [inline webapp/login/views.py:96]: get the stores and only store whether there are any

File: webapp/login/views.py [hunk 4]
```diff
@@ -189,8 +187,8 @@ def login_callback():
     }
 
     if logic.get_stores(account["stores"], roles=["admin", "review", "view"]):
-        flask.session["publisher"]["stores"] = admin_api.get_stores(
-            flask.session
+        flask.session["publisher"]["has_stores"] = (
+            len(admin_api.get_stores(flask.session)) > 0
```
Comment (Lukewh) [inline webapp/login/views.py:191]: get the stores and only store whether there are any

File: webapp/publisher/snaps/logic.py [hunk 5]
```diff
@@ -400,8 +400,8 @@ def categorise_media(media):
     return icon_urls, screenshot_urls, banner_urls
 
 
-def get_store_name(store_id, user):
-    available_stores = filter_available_stores(user["stores"])
+def get_store_name(store_id, stores):
+    available_stores = filter_available_stores(stores)
```
Comment (Lukewh) [inline webapp/publisher/snaps/logic.py:404]: pass and use the stores list directly

File: webapp/publisher/snaps/views.py [hunk 6]
```diff
@@ -286,9 +290,9 @@ def redirect_get_register_name():
 @publisher_snaps.route("/register-snap")
 @login_required
 def get_register_name():
-    user = publisher_api.get_account(flask.session)
+    stores = admin_api.get_stores(flask.session)
```
Comment (Lukewh) [inline webapp/publisher/snaps/views.py:293]: The account no longer has the list of stores, so get the list of stores directly from the API

File: webapp/publisher/snaps/views.py [hunk 7]
```diff
@@ -351,9 +355,9 @@ def post_register_name():
             registrant_comment=registrant_comment,
         )
     except StoreApiResponseErrorList as api_response_error_list:
-        user = publisher_api.get_account(flask.session)
+        stores = admin_api.get_stores(flask.session)
```
Comment (Lukewh) [inline webapp/publisher/snaps/views.py:358]: The account no longer has the list of stores, so get the list of stores directly from the API

File: webapp/publisher/snaps/views.py [hunk 8]
```diff
@@ -451,11 +455,11 @@ def post_register_name_json():
 @publisher_snaps.route("/register-name-dispute")
 @login_required
 def get_register_name_dispute():
-    user = publisher_api.get_account(flask.session)
+    stores = admin_api.get_stores(flask.session)
```
Comment (Lukewh) [inline webapp/publisher/snaps/views.py:458]: The account no longer has the list of stores, so get the list of stores directly from the API

File: webapp/publisher/snaps/views.py [hunk 9]
```diff
@@ -496,11 +500,11 @@ def post_register_name_dispute():
 @publisher_snaps.route("/request-reserved-name")
 @login_required
 def get_request_reserved_name():
-    user = publisher_api.get_account(flask.session)
+    stores = admin_api.get_stores(flask.session)
```
Comment (Lukewh) [inline webapp/publisher/snaps/views.py:503]: The account no longer has the list of stores, so get the list of stores directly from the API

**Review by steverydz — APPROVED** [review]

**Pull request merged**